hathor/indexes/address_index.py

```diff
@@ -25,11 +26,17 @@
 logger = get_logger()
 
 
-class AddressIndex(ABC):
+class AddressIndex(BaseIndex):
     """ Index of inputs/outputs by address
     """
     pubsub: Optional['PubSubManager']
 
+    def init_loop_step(self, tx: BaseTransaction) -> None:
+        tx_meta = tx.get_metadata()
+        if tx_meta.voided_by:
```

Comment:
Why do you skip voided transactions? What about soft voided ones?

Reply:
As far as I understand, this index is not interested in soft voided transactions, even if they are soft voided.

Reply:
As far as I understood, this index has all transactions, including the voided ones.

Reply:
Sorry, you're right! I'll add a test to catch this bug and fix it.

Reply:
Have you added it?

Reply:
Thanks!